### DISCUSSION TITLE
Query Variable with key value like Custom Variable

### DISCUSSION THREAD

**Discussion by devamin** [grafana/grafana#45854] [Data Sources]
Having the possibility to create a query variable with key value pair is a good feature which is already implemented in Custom Variable.
e.g. for mysql data source

```
SELECT id as `key`, something as value
FROM table
```

**Comment by Elyahou** [comment]
This is already available:

```
SELECT hostname AS __text, id AS __value FROM my_host
```

**Reply by devamin** [reply]
Thanks perfect

**Comment by gabor** [comment]
closing, as this seems to solve the problem.
